src/node_perf.cc

```diff
@@ -373,6 +374,47 @@ void Timerify(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(wrap);
 }
 
+static void on_uv_interval(uv_timer_t* req) {
+  delay_timer_t* delay = reinterpret_cast<delay_timer_t*>(req->data);
+
+  uint64_t time = uv_hrtime();
+  uint64_t delta = time - delay->prev;
+
+  if (delta > DELAY_TIMER_THRESHOLD_NS) {
+    delta = (delta - DELAY_TIMER_RESOLUTION_NS) / 1e6;
+    if (delta > 0xffffffff) delta = 0xffffffff;
+    delay->delay += (uint32_t) delta;
```

Comment:
```suggestion
    delay->delay += static_cast<uint32_t>(delta);
```

Or maybe make the value `uint64_t`? That way we don’t have to worry about a chance of overflow in (very long-running) applications.

Reply:
Unless there is seriously something very very wrong with the application, a delta that does not fit within a uint32_t would be exceedingly unlikely and I'm not certain that requiring a BigNum on the API side would be worthwhile.